jackson2/src/main/java/org/jdbi/v3/jackson2/Jackson2Plugin.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.jackson2;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.jdbi.v3.core.Jdbi;
+import org.jdbi.v3.core.spi.JdbiPlugin;
+import org.jdbi.v3.meta.Beta;
+
+/**
+ * Jackson 2 integration plugin.
+ *
+ * Adds support for {@code @Json} qualifying annotation.
+ *
+ * @see Json
+ */
+@Beta
+public class Jackson2Plugin implements JdbiPlugin {
+
+    private final ObjectMapper mapper;
+
+    public Jackson2Plugin() {
+        this(new ObjectMapper());
+    }
+
+    public Jackson2Plugin(ObjectMapper mapper) {
```

Comment:
kinda feels like this is different from how the other plugins work, delegating all their configuration to Config classes alone instead of having special constructors that will eventually become a burden when more configuration is added

Reply:
I'm ambivalent on this one.

I do agree that it's a departure from how other plugins work, so maybe it's worth making it consistent for consistency's sake.

Reply:
@TheRealMarnes by "ugly mess" do you mean overloading constructors for more config parameters? I don't see that happening. The `ObjectMapper` is the focal point of this module so it's kind of special in that sense.

Reply:
That's what they always say at first :)

But yeah, the sum of _maybe_ wanting to add more in the future because of expansion and the deviation from an established pattern, makes me say we shouldn't do this.

Reply:
On reflection I think I agree with Marnes on this.

We can always change our mind in the future and put this constructor back, but we can't take it back once it's published API.

Reply:
I never intended to continue adding new constructors if there's any more configuration to do.  I only had this constructor because it really is an integral part of the integration, and currently there's no need for further configuration - so you don't need to know about the config class at all.  I guess consistency is an argument, but I do think ObjectMapper is integral to this module in a way that e.g. changing null binding or configuring case mapping isn't.

Reply:
I really don't see this constructor as added value over the config class, especially because we can provide a default anyway...

If we go ahead with splitting jackson out of the feaure itself as discussed further below though, then this constructor would become a necessity: you could have a `new JacksonPlugin()` or `new GsonPlugin()`, but would require a `new JsonPlugin(mapperImpl)`.

Reply:
I did this.